src/core/jbig2.js

```diff
@@ -1306,6 +2103,62 @@ var Jbig2Image = (function Jbig2ImageClosure() {
   return Jbig2Image;
 })();
 
+/**
+ * For JBIG2's we use a library to decode these images and
+ * the stream behaves like all the other DecodeStreams.
+ */
+var Jbig2Stream = (function Jbig2StreamClosure() {
+  function Jbig2Stream(stream, maybeLength, dict, params) {
+    this.stream = stream;
+    this.maybeLength = maybeLength;
+    this.dict = dict;
+    this.params = params;
+
+    DecodeStream.call(this, maybeLength);
+  }
+
+  Jbig2Stream.prototype = Object.create(DecodeStream.prototype);
+
+  Object.defineProperty(Jbig2Stream.prototype, 'bytes', {
+    get: function Jbig2Stream_bytes() {
+      // If this.maybeLength is null, we'll get the entire stream.
+      return shadow(this, 'bytes', this.stream.getBytes(this.maybeLength));
+    },
+    configurable: true,
+  });
+
+  Jbig2Stream.prototype.ensureBuffer = function Jbig2Stream_ensureBuffer(req) {
+    if (this.bufferLength) {
+      return;
+    }
+
+    var jbig2Image = new Jbig2Image();
+
+    var chunks = [];
+    if (isDict(this.params)) {
+      var globalsStream = this.params.get('JBIG2Globals');
+      if (isStream(globalsStream)) {
+        var globals = globalsStream.getBytes();
+        chunks.push({ data: globals, start: 0, end: globals.length, });
+      }
+    }
+    chunks.push({ data: this.bytes, start: 0, end: this.bytes.length, });
+    var data = jbig2Image.parseChunks(chunks);
+    var dataLength = data.length;
+
+    // JBIG2 had black as 1 and white as 0, inverting the colors
+    for (var i = 0; i < dataLength; i++) {
+      data[i] ^= 0xFF;
+    }
+
+    this.buffer = data;
+    this.bufferLength = dataLength;
+    this.eof = true;
+  };
+
+  return Jbig2Stream;
+})();
+
 export {
-  Jbig2Image,
```

Comment:
Since there's people that use the image decoders in PDF.js in other projects, it would probably be a good idea to keep exporting `Jbig2Image` here (and we may also want to update https://github.com/notmasteryet/jpgjs/ in the future).